### PR TITLE
Sshdisablecommunication

### LINKED ISSUE #658: Option for disable SSH communication with firewalls.
Currently we have two options for the communication with a firewall or cluster's nodes: SSH or FWCloud-Agent. 
For the MyFWCloud service we will disable the SSH mechanism for security reasons.

For make it possible we need to do the next thins in the FWCloud-API:
- [x] New configuration option for disable communications by means of SSH.
- [x] If this option is enabled, every API request that requires communication with firewalls or nodes of a cluster will generate an error if such communication is by means of SSH.
- [x] This error must be communicated to FWCloud-UI and it will show a message about it to the user.
- [ ] Create software tests for verify this in all API calls that require communication with firewalls.

### PR DESCRIPTION
**Pull request by soltecsis-iortega**
Closes #658

### CODE REVIEW

File: src/communications/ssh.communication.ts [hunk 1]
```diff
@@ -171,6 +194,9 @@ export class SSHCommunication extends Communication<SSHConnectionData> {
 
     async getRealtimeStatus(statusFilepath: string): Promise<string> {
         try {
+            if(!app().config.get('session.ssh_enable')) {
```
Comment (soltecsis-jurios) [inline src/communications/ssh.communication.ts:197]: Why don't use `this.getConfigValue()`?

File: src/communications/ssh.communication.ts [hunk 2]
```diff
@@ -75,6 +84,9 @@ export class SSHCommunication extends Communication<SSHConnectionData> {
 
     async installOpenVPNClientConfigs(dir: string, configs: {name: string, content: string}[], eventEmitter: EventEmitter = new EventEmitter()): Promise<void> {
         try {
+            if(!app().config.get('session.ssh_enable')) {
```
Comment (soltecsis-jurios) [inline src/communications/ssh.communication.ts:87]: Why don't use `this.getConfigValue()`?

File: src/communications/ssh.communication.ts [hunk 3]
```diff
@@ -47,6 +53,9 @@ export class SSHCommunication extends Communication<SSHConnectionData> {
 
     async installOpenVPNServerConfigs(dir: string, configs: {name: string, content: string}[], eventEmitter?: EventEmitter): Promise<void> {
         try {
+            if(!app().config.get('session.ssh_enable')) {
```
Comment (soltecsis-jurios) [inline src/communications/ssh.communication.ts:56]: Why don't use `this.getConfigValue()`?

File: src/communications/ssh.communication.ts [hunk 4]
```diff
@@ -20,6 +23,9 @@ export class SSHCommunication extends Communication<SSHConnectionData> {
 
     async installFirewallPolicy(scriptPath: string, eventEmitter: EventEmitter = new EventEmitter()): Promise<string> {
         try {
+            if(!app().config.get('session.ssh_enable')) {
```
Comment (soltecsis-jurios) [inline src/communications/ssh.communication.ts:26]: Why don't use `this.getConfigValue()`?

File: src/controllers/firewalls/firewall.controller.ts [hunk 5]
```diff
@@ -236,4 +237,38 @@ export class FirewallController extends Controller {
             throw error;
         }
     }
+
+    @Validate()
+    async configCommunication(request: Request): Promise<ResponseBuilder> {
+        const input = request.body;
+
+        (await FirewallPolicy.ping(this._fwCloud,request.session.user)).authorize();
+        //Hay que tener cuidado de como se obtienen los datos y demas pero bueno ya estaria
```
Comment (soltecsis-jurios) [inline src/controllers/firewalls/firewall.controller.ts:246]: Remove comment

File: src/controllers/firewalls/firewall.controller.ts [hunk 6]
```diff
@@ -236,4 +237,38 @@ export class FirewallController extends Controller {
             throw error;
         }
     }
+
+    @Validate()
+    async configCommunication(request: Request): Promise<ResponseBuilder> {
```
Comment (soltecsis-jurios) [inline src/controllers/firewalls/firewall.controller.ts:242]: We can provide this value when the UI is getting the data about the `fwcloud` (as this config value is `fwcloud`-related config parameter) and save this extra request.

**Review by soltecsis-jurios — CHANGES_REQUESTED** [review]

**Review by cmunyoz — APPROVED** [review]

**Pull request merged**